src/corefx/System.Globalization.Native/localeStringData.cpp

```diff
@@ -240,3 +255,83 @@ extern "C" int32_t GetLocaleInfoString(const UChar* localeName, LocaleStringData
 
 	return UErrorCodeToBool(status);
 }
+
+/*
+Function:
+NormalizeTimePattern
+
+Convert an ICU non-localized time pattern to .NET format
+*/
+void NormalizeTimePattern(const UnicodeString *srcPattern, UnicodeString *destPattern)
```

Comment:
Sure put in managed if the native side has any sort of non-trivial logic or possible maintenance. Unicodestring makes strings pretty easy in native code IMO